### DISCUSSION TITLE
Lesson 6: forge script error

### DISCUSSION THREAD

**Discussion by guyInTheChair** [Cyfrin/foundry-full-course-cu#82] [Q&A]
Hello 

While running the below command in terminal I am getting an error. This is from Lesson 6. Could anyone please help?

forge script script/DeploySimpleStorage.s.sol 

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/33867823/47c3840a-3515-4968-8f3e-b9951538035e)

Below is my DeploySimpleStorage.s.sol code
```solidity
// SPDX-License-Identifier: MIT

pragma solidity 0.8.19;

import {Script} from "forge-std/Script.sol";
import {SimpleStorage} from "../src/SimpleStorage.sol";

contract DeploySimpleStorage is Script {
    function run() external returns (SimpleStorage) {
        vm.startBroadcast();

        SimpleStorage simpleStorage = new SimpleStorage();

        vm.stopBroadcast();
        return simpleStorage;
    }
}
```

**Comment by cromewar** [comment]
Hello @guyInTheChair this is a strange problem as your cost is exactly the same as on the Repo.
I have some suggest for you tough:

1. add override to your function as follows: `function run() external override returns (SimpleStorage) {`
2. I that does not work try changing the version of the compiler/solidity version.
3.  Ensure you're executing the command forge scripts/DeploySimpleStorage.s.sol in the correct directory where the DeploySimpleStorage.s.sol file is located.

**Reply by guyInTheChair** [reply]
Hey! Will try these and let you know whether it worked or not.

**Reply by CodeTivity08** [reply]
Mine gave a different error message:
```
Error: 
Compiler run failed:
Error (7858): Expected pragma, import directive or contract/interface/library/struct/enum/constant/function/error definition.
  --> script/DeploySimpleStorage.s.sol:16:1:
   |
16 | }
   | ^
```

ChatGPT tells me this:
"The error is caused by an incomplete import statement. It seems that the SimpleStorage contract is not being imported correctly.

Please make sure that the path to the SimpleStorage.sol file is correct. Double-check that the file is located in the correct directory relative to the DeploySimpleStorage.sol file.

Assuming that the SimpleStorage.sol file is in the ../src/ directory, you should update the import statement as follows:

solidity
Copy code
import {SimpleStorage} from "../src/SimpleStorage.sol";
Make sure the path is correct and try compiling the code again."

That was yesterday. Now I run the exact script (unchanged from yesterday) again, and it compiles successfully. This is the second time I had to quit the program and than continue the next day, and it suddenly works.

Any thoughts as to why?

**Reply by save-the-planet08** [reply]
I had the same problem and because of this I tried it another time and then it worked

**Comment by alymurtazamemon** [comment]
@guyInTheChair Did you save the file?

**Reply by guyInTheChair** [reply]
Yes I did save the file :)

**Comment by CodeTivity08** [comment]
Mine gave a different error message:
```
Error: 
Compiler run failed:
Error (7858): Expected pragma, import directive or contract/interface/library/struct/enum/constant/function/error definition.
  --> script/DeploySimpleStorage.s.sol:16:1:
   |
16 | }
   | ^
```

ChatGPT tells me this:
"The error is caused by an incomplete import statement. It seems that the SimpleStorage contract is not being imported correctly.

Please make sure that the path to the SimpleStorage.sol file is correct. Double-check that the file is located in the correct directory relative to the DeploySimpleStorage.sol file.

Assuming that the SimpleStorage.sol file is in the ../src/ directory, you should update the import statement as follows:

solidity
Copy code
import {SimpleStorage} from "../src/SimpleStorage.sol";
Make sure the path is correct and try compiling the code again."

That was yesterday. Now I run the exact script (unchanged from yesterday) again, and it compiles successfully. This is the second time I had to quit the program and than continue the next day, and it suddenly works.

Any thoughts as to why?
